Sources/TensorFlow/Core/LazyTensorTFFunctionBuilder.swift

```diff
@@ -0,0 +1,315 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import CTensorFlow
+
+class TFGraph {
+    /// The `TF_Operation *` type.
+    typealias CTFOperation = OpaquePointer
+
+    enum Input {
+        case single(TF_Output)
+        case list([TF_Output])
+    }
+
+    /// The pointer to the underlying TF Graph.
+    let cTFGraph: CTFGraph = TF_NewGraph()
+    var inputs: [TF_Output] = []
+    var nodes: [CTFOperation?] = []
+    var outputs: [TF_Output] = []
+    var outputGroups: [Int] = []
```

Comment:
This is an array of the sizes of the output groups, right? If so, I think `outputGroupCounts` would be a clearer name.

Or maybe even replace `var outputs: [TF_Output]` and `var outputGroupCounts: [Int]` with `var outputGroups: [[TF_Output]]` -- unless there's some reason why the `var outputs: [TF_Output]` and `var outputGroupCounts: [Int]` representation is more convenient for the operations you are doing.

Reply:
Changed the property name, but left them as separate. `[TF_Output]` is convenient to interact the TF Eager runtime.